sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -82,6 +82,65 @@ case class GpuCreateArray(children: Seq[Expression], useStringTypeWhenEmpty: Boo
   }
 }
 
+case class GpuCreateMap(children: Seq[Expression], useStringTypeWhenEmpty: Boolean)
+    extends GpuExpression {
+
+  private val valueIndices: Seq[Int] = children.indices.filter(_ % 2 != 0)
+  private val keyIndices: Seq[Int] = children.indices.filter(_ % 2 == 0)
+
+  lazy val keys: Seq[Expression] = keyIndices.map(children)
+  lazy val values: Seq[Expression] = valueIndices.map(children)
+
+  private val defaultElementType: DataType = {
+    if (useStringTypeWhenEmpty) {
+      StringType
+    } else {
+      NullType
+    }
+  }
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    withResource(new Array[ColumnVector](children.size)) { columns =>
+      val numRows = batch.numRows()
+      children.indices.foreach { index =>
+        columns(index) = GpuExpressionsUtils.columnarEvalToColumn(children(index), batch).getBase
+      }
+
+      if (children.length == 2) {
+        withResource(ColumnVector.makeStruct(columns: _*)) { struct =>
+          GpuColumnVector.from(ColumnVector.makeList(numRows, DType.STRUCT, struct), dataType)
+        }
+      } else {
+        val structs = Range(0, columns.length, 2)
```

Comment:
Does this code not work if `children.length == 2`? It looks like it would and then we could remove the conditional and the extra code.

Reply:
Yes, this code works in both cases. I will remove the special case.